src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -649,6 +643,64 @@ - (void)updateWithPeripheral:(CBPeripheral *)peripheral
     [self stopScanning];
 }
 
+- (void)addPeripheralToCache:(CBPeripheral *)peripheral data:(NSData *)data
+{
+    dispatch_source_t timeoutTimer;
```

Comment:
Should this bail out early if `_cachedPeripherals` is nil?  Or is it never expected to be nil here?

Reply:
It is not supposed to be nil. (which was a wrong assertion with `removePeripheralsFromCache` as you mentioned).